### ISSUE TITLE
libcxx test fails on armv7-linux

### ISSUE THREAD

**Issue by rovka** [llvm/llvm-project#53848]
std/language.support/cmp/cmp.alg/strong_order_long_double.verify.cpp fails with:
```
error: 'error' diagnostics expected but not seen: 
  File * Line * (directive at /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/llvm-project/libcxx/test/std/language.support/cmp/cmp.alg/strong_order_long_double.verify.cpp:30): std::strong_order is unimplemented for this floating-point type
1 error generated.

```
Full output attached:
[strong_order.txt](https://github.com/llvm/llvm-project/files/8068945/strong_order.txt)

**Comment by DavidSpickett** [comment]
From the test:
```
// The following platforms have sizeof(long double) == sizeof(double), so this test doesn't apply to them.
// UNSUPPORTED: target={{arm64|armv8|armv7|armv7m|powerpc|powerpc64}}-{{.+}}
```

For the buildbot (which actually runs on v8 hardware) we set:
```
set(LIBCXX_TARGET_TRIPLE "armv7-linux-gnueabihf" CACHE STRING "")
```

So it'll match "armv7-..." I wonder if in the native build the target is "armhf" instead and so this doesn't catch it.

**Comment by rovka** [comment]
The full output dumped by the test looks like this:

>  "/home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/Phase3/Release/llvmCore-14.0.0-rc1.obj/bin/clang-14" -cc1 **-triple armv7-unknown-linux-gnueabihf** -fsyntax-only -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name strong_order_long_double.verify.cpp -mrelocation-model static -mframe-pointer=all -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -target-cpu generic -target-feature +vfp2 -target-feature +vfp2sp -target-feature +vfp3 -target-feature +vfp3d16 -target-feature +vfp3d16sp -target-feature +vfp3sp -target-feature -fp16 -target-feature -vfp4 -target-feature -vfp4d16 -target-feature -vfp4d16sp -target-feature -vfp4sp -target-feature -fp-armv8 -target-feature -fp-armv8d16 -target-feature -fp-armv8d16sp -target-feature -fp-armv8sp -target-feature -fullfp16 -target-feature +fp64 -target-feature +d32 -target-feature +neon -target-feature -sha2 -target-feature -aes -target-feature -fp16fml -target-abi aapcs-linux -mfloat-abi hard -fallow-half-arguments-and-returns -mllvm -treat-scalable-fixed-error-as-warning -debugger-tuning=gdb -v -fcoverage-compilation-dir=/home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/Phase3/Release/llvmCore-14.0.0-rc1.obj/runtimes/runtimes-bins/libcxx/test/std/language.support/cmp/cmp.alg -nostdinc++ -resource-dir /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/Phase3/Release/llvmCore-14.0.0-rc1.obj/lib/clang/14.0.0 -I /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/Phase3/Release/llvmCore-14.0.0-rc1.obj/include/armv7l-unknown-linux-gnueabihf/c++/v1 -I /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/Phase3/Release/llvmCore-14.0.0-rc1.obj/include/c++/v1 -I /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/Phase3/Release/llvmCore-14.0.0-rc1.obj/runtimes/runtimes-bins/libcxx/include/c++build -D __STDC_FORMAT_MACROS -D __STDC_LIMIT_MACROS -D __STDC_CONSTANT_MACROS -I /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/llvm-project/libcxx/test/support -D _LIBCPP_DISABLE_AVAILABILITY -D _LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D _LIBCPP_HAS_THREAD_API_PTHREAD -D _LIBCPP_ABI_VERSION=1 -D _LIBCPP_HAS_NO_INCOMPLETE_FORMAT -D _LIBCPP_HAS_NO_INCOMPLETE_RANGES -internal-isystem /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/Phase3/Release/llvmCore-14.0.0-rc1.obj/lib/clang/14.0.0/include -internal-isystem /usr/local/include -internal-isystem /usr/lib/gcc/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/include -internal-externc-isystem /usr/include/arm-linux-gnueabihf -internal-externc-isystem /include -internal-externc-isystem /usr/include -Werror -Wall -Wextra -Wshadow -Wundef -Wno-unused-command-line-argument -Wno-attributes -Wno-pessimizing-move -Wno-c++11-extensions -Wno-user-defined-literals -Wno-noexcept-type -Wno-atomic-alignment -Wsign-compare -Wunused-variable -Wunused-parameter -Wunreachable-code -Wno-unused-local-typedef -Werror=thread-safety -Wuser-defined-warnings -Wno-macro-redefined -Wno-macro-redefined -Wno-macro-redefined -Wno-macro-redefined -Wno-error -std=c++2b -fdeprecated-macro -fdebug-compilation-dir=/home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/Phase3/Release/llvmCore-14.0.0-rc1.obj/runtimes/runtimes-bins/libcxx/test/std/language.support/cmp/cmp.alg -ferror-limit 0 -fcoroutines-ts -fno-signed-char -fgnuc-version=4.2.1 -fcxx-exceptions -fexceptions -verify -verify-ignore-unexpected=note -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -x c++ /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/llvm-project/libcxx/test/std/language.support/cmp/cmp.alg/strong_order_long_double.verify.cpp
> clang -cc1 version 14.0.0 based upon LLVM 14.0.0 **default target armv7l-unknown-linux-gnueabihf**
> ignoring nonexistent directory "/usr/lib/gcc/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/include"
> ignoring nonexistent directory "/include"
> #include "..." search starts here:
> #include <...> search starts here:
>  /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/Phase3/Release/llvmCore-14.0.0-rc1.obj/include/armv7l-unknown-linux-gnueabihf/c++/v1
>  /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/Phase3/Release/llvmCore-14.0.0-rc1.obj/include/c++/v1
>  /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/Phase3/Release/llvmCore-14.0.0-rc1.obj/runtimes/runtimes-bins/libcxx/include/c++build
>  /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/llvm-project/libcxx/test/support
>  /home/tcwg-buildslave/workspace/tcwg-llvm-release/tcwg-tk1_32-build/rc1/Phase3/Release/llvmCore-14.0.0-rc1.obj/lib/clang/14.0.0/include
>  /usr/local/include
>  /usr/include/arm-linux-gnueabihf
>  /usr/include
> End of search list.

To me it looks like the triple passed to clang would make the test UNSUPPORTED, but I'm not sure if that's the triple that's being matched against (would have to dig a bit more to convince myself).

**Comment by DavidSpickett** [comment]
If you have the build around still try changing the test to:
```
// UNSUPPORTED: target={{arm64|armv8|armv7(m|l)?|powerpc|powerpc64}}-{{.+}}
```

Then it'll pick up armv7/armv7l/armv7m.

I should change the v7 buildbots to use the `v7l` triple as well, to stop this happening in future.

**Comment by DavidSpickett** [comment]
To do that, at least on main: https://reviews.llvm.org/D119948

**Comment by DavidSpickett** [comment]
The above change is in, it could be picked to the release if you want. Otherwise I think we can safely say this isn't an actual bug in libcxx or llvm in general.

**Comment by rovka** [comment]
Thanks @DavidSpickett! I think this is worth cherry-picking.
